source/base/incremental_function.cc

```diff
@@ -0,0 +1,101 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2007 - 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/incremental_function.h>
+
+#include <deal.II/lac/vector.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Functions
+{
+  template <int dim, typename RangeNumberType>
+  IncrementalFunction<dim, RangeNumberType>::IncrementalFunction(
+    Function<dim, RangeNumberType> &base)
+    : base(base)
+    , delta_t(time_type())
```

Comment:
maybe use `numbers::signaling_NaN()` to denote the invalid value?

Reply:
i think zero by default is ok.

Reply:
I think that @bangerth has a point. If someone forgets to call `set_decrement()` then all output values will be zero. It would not necessarily be intuitive as to why this is happening. What do you think?

Reply:
agree, will change.